core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1680,6 +1688,61 @@ private ObjList<Function> generateCastFunctions(
         return castFunctions;
     }
 
+    private RecordCursorFactory generateFill(QueryModel model, RecordCursorFactory groupByFactory, SqlExecutionContext executionContext) throws SqlException {
+        if (!isFillStridePresent(model.getNestedModel())) {
+            return groupByFactory;
+        } else {
+            try {
+                final QueryModel nested = model.getNestedModel();
+                Function fillFromFunc = TimestampConstant.NULL;
+                Function fillToFunc = TimestampConstant.NULL;
+                final ExpressionNode fillFrom = nested.getFillFrom();
+                final ExpressionNode fillTo = nested.getFillTo();
+                final ExpressionNode fillStride = nested.getFillStride();
+
+                ObjList<ExpressionNode> fillValuesExprs = nested.getFillValue();
+                ObjList<Function> fillValues = new ObjList<>(fillValuesExprs.size());
+
+                ExpressionNode expr;
+                for (int i = 0, n = fillValuesExprs.size(); i < n; i++) {
+                    expr = fillValuesExprs.getQuick(0);
+                    if (isNoneKeyword(expr.token)) {
+                        return groupByFactory;
+                    }
+                    final Function fillValueFunc = functionParser.parseFunction(expr, EmptyRecordMetadata.INSTANCE, executionContext);
+                    fillValues.add(fillValueFunc);
+                }
+
+                if (fillValues.size() == 0) {
+                    return groupByFactory;
+                }
+
+                if (fillValues.size() == 1 && isNoneKeyword(fillValues.getQuick(0).getName())) {
+                    return groupByFactory;
+                }
+
+                if (fillFrom != null) {
+                    fillFromFunc = functionParser.parseFunction(fillFrom, EmptyRecordMetadata.INSTANCE, executionContext);
+                    coerceRuntimeConstantType(fillFromFunc, ColumnType.TIMESTAMP, executionContext, "from lower bound must be a constant expression convertible to a TIMESTAMP", -1);
```

Comment:
nit: we should include fillFrom's position into the returned error. Same with fillTo.